JSONArray.java

```diff
@@ -154,7 +154,7 @@ public JSONArray(String source) throws JSONException {
      *            A Collection.
      */
     public JSONArray(Collection<?> collection) {
-        this.myArrayList = new ArrayList<Object>();
+        this.myArrayList = new ArrayList<Object>(collection == null ? 0 : collection.size());
```

Comment:
My thought was that if you pass null, you are probably trying to create an empty JSONArray. But for backwards compatibility it may be better to use the standard capacity.  Personally if I'm using this constructor it's because I've already finalized my object somewhere else and I'm just preparing it for serialization. But that generalization may not be what we want to imply.

I'll go ahead and change it.